src/diffusers/loaders/single_file_utils.py

```diff
@@ -442,6 +330,76 @@ def infer_model_type(original_config, checkpoint, model_type=None):
     return model_type
 
 
+def is_clip_model(checkpoint):
+    if CHECKPOINT_KEY_NAMES["clip"] in checkpoint:
+        return True
+
+    return False
+
+
+def is_open_clip_model(checkpoint):
+    if CHECKPOINT_KEY_NAMES["open_clip"] in checkpoint:
+        return True
+
+    return False
+
+
+def is_open_clip_sdxl_model(checkpoint):
+    if CHECKPOINT_KEY_NAMES["open_clip_sdxl"] in checkpoint:
+        return True
+
+    return False
+
+
+def is_clip_model_in_single_file(class_obj, checkpoint):
+    is_clip_in_checkpoint = any(
+        [is_clip_model(checkpoint), is_open_clip_model(checkpoint), is_open_clip_sdxl_model(checkpoint)]
+    )
+    if (
+        class_obj.__name__ == "CLIPTextModel" or class_obj.__name__ == "CLIPTextModelWithProjection"
+    ) and is_clip_in_checkpoint:
+        return True
+
+    return False
+
+
+def fetch_diffusers_config(checkpoint, subfolder=None):
+    if (
+        CHECKPOINT_KEY_NAMES["inpainting"] in checkpoint
+        and checkpoint[CHECKPOINT_KEY_NAMES["inpainting"]].shape[1] == 9
+    ):
+        if CHECKPOINT_KEY_NAMES["v2"] in checkpoint and checkpoint[CHECKPOINT_KEY_NAMES["v2"]].shape[-1] == 1024:
+            model_config = DIFFUSERS_DEFAULT_CONFIGS["inpainting_v2"]
+        else:
+            model_config = DIFFUSERS_DEFAULT_CONFIGS["inpainting"]
+
+    elif CHECKPOINT_KEY_NAMES["v2"] in checkpoint and checkpoint[CHECKPOINT_KEY_NAMES["v2"]].shape[-1] == 1024:
+        model_config = DIFFUSERS_DEFAULT_CONFIGS["v2"]
+
+    elif CHECKPOINT_KEY_NAMES["playground-v2-5"] in checkpoint:
+        model_config = DIFFUSERS_DEFAULT_CONFIGS["playground-v2-5"]
+
+    elif CHECKPOINT_KEY_NAMES["xl_base"] in checkpoint:
+        model_config = DIFFUSERS_DEFAULT_CONFIGS["xl_base"]
+
+    elif CHECKPOINT_KEY_NAMES["xl_refiner"] in checkpoint:
+        model_config = DIFFUSERS_DEFAULT_CONFIGS["xl_refiner"]
+
+    elif CHECKPOINT_KEY_NAMES["upscale"] in checkpoint:
+        model_config = DIFFUSERS_DEFAULT_CONFIGS["upscale"]
+
+    elif CHECKPOINT_KEY_NAMES["controlnet"] in checkpoint:
+        model_config = DIFFUSERS_DEFAULT_CONFIGS["controlnet"]
+
+    else:
+        model_config = DIFFUSERS_DEFAULT_CONFIGS["v1"]
+
+    if subfolder:
+        model_config["subfolder"] = subfolder
+
+    return model_config
```

Comment:
It should either return the contents of `model_index.json` or the `config.json` no?

Reply:
Ah nevermind. You're leveraging `load_config()` already.